src/core/crypto/KeyPair.ts

```diff
@@ -74,21 +66,16 @@ export class KeyPair {
      * @param {module:crypto/keyPair~KeyPair} keyPair The key pair for which to create the shared key.
      * @param {Uint8Array} publicKey The public key for which to create the shared key.
      * @param {Uint8Array} salt A salt that should be applied to the shared key.
-     * @param {SignSchema} signSchema The Sign Schema. (KECCAK_REVERSED_KEY / SHA3)
+     * @param {networkType} networkType Catapult network identifier
      * @returns {Uint8Array} The shared key.
      */
-    public static deriveSharedKey = (keyPair, publicKey, salt, signSchema = SignSchema.SHA3) => {
+    public static deriveSharedKey = (keyPair, publicKey: Uint8Array, salt: Uint8Array, networkType: NetworkType) => {
         if (Utility.Key_Size !== salt.length) {
             throw Error(`salt has unexpected size: ${salt.length}`);
         }
         if (Utility.Key_Size !== publicKey.length) {
             throw Error(`public key has unexpected size: ${salt.length}`);
         }
-        let secretKey = keyPair.privateKey;
-        // KECCAK_REVERSED_KEY uses reversed private key.
-        if (signSchema === SignSchema.KECCAK_REVERSED_KEY) {
-            secretKey = convert.hexToUint8Reverse(convert.uint8ToHex(secretKey));
-        }
-        return Utility.catapult_crypto.deriveSharedKey(salt, secretKey, publicKey, Utility.catapult_hash.func, signSchema);
+        return Utility.catapult_crypto.deriveSharedKey(salt, keyPair.privateKey, publicKey, Utility.catapult_hash.func, networkType);
```

Comment:
I would keep the reverse hooks, maybe create a utility method that reverse if necessary (called every time there was a hexToUint8Reverse). Just in case you want to add the reverse stuff in the future.

Reply:
Not sure about this. Let's confirm with core first. If the decision is non-reversal I don't think we should keep it. 

The 'hexToUint8Reverse' definitely stays, as will probably need it at some point. E.g. the UInt64 is using `Little Endian`.